### DISCUSSION TITLE
Too many arguments in find_similar core function

### DISCUSSION THREAD

**Discussion by quillcraftsman** [findsimilar/find-similar#31] [General]
```python
def find_similar(  # pylint: disable=too-many-arguments
    text_to_check,
    texts,
    language="russian",
    count=5,
    dictionary=None,
    remove_stopwords=True,
) -> list[TokenText]:
```

[R0913 pylint](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/too-many-arguments.html)
 
Too many arguments 6 (maximum 5 in pylint). It means that interface became too big and complex.
I propose to think about this problem. 
* Why interface become too complex? 
* What can we do to fix it? 
* Or it's normal and nothing to fix?

**Comment by quillcraftsman** [comment]
I think this way: if we want to use `find_similar` as universal function witch will work with any objects (not only texts), the interface should be:
```python
def find_similar( 
    object_to_check,
    object_list,
    count=5,
    dictionary=None, # May be if it will be some map where one ore many objects related with one or many
) -> list:
```

`language` and `remove_stopwords` parameters about texts only. That's why these parameters should be removed. 
May be we can send one additional parameter (callback, or class, or instance) to set object types and this additional parameters (like `language` and `remove_stopwords`)

**Reply by quillcraftsman** [reply]
But may be it's **Premature Optimization**. The whole project works with texts only at this moment.

**Reply by vtyushkevich** [reply]
May be we should to create class with variables: language, count, dictionary, remove_stopwords and send instance of this class to core functions?
And to leave arguments with text for now as is. We could return to the question "objects or text" later

**Comment by quillcraftsman** [comment]
> May be we should to create class with variables: language, count, dictionary, remove_stopwords and send instance of this class to core functions? And to leave arguments with text for now as is. We could return to the question "objects or text" later

Sounds reasonable. If we change main function interface we should create new release 2.0 without
compatibility. We need to be prepared for this. `find-similar` used in demo API.
